### PR TITLE
[DW-159] Adding website document types to local search

### PR DESCRIPTION
**Pull request by giacomolm**

### CODE REVIEW

File: repository-data/application/src/main/resources/hcm-content/content/documents/administration/website/search-tab-labels.yaml [hunk 1]
```diff
@@ -0,0 +1,70 @@
+---
```
Comment (wojtek-oledzki) [inline repository-data/application/src/main/resources/hcm-content/content/documents/administration/website/search-tab-labels.yaml]: just add new labels to existing "administration/searchtab-labels"

File: site/src/main/java/uk/nhs/digital/common/enums/SearchArea.java [hunk 2]
```diff
@@ -4,10 +4,12 @@
     ALL,
     DATA,
     SERVICES,
-    NEWS;
+    NEWS,
```
Comment (jacsamell) [inline site/src/main/java/uk/nhs/digital/common/enums/SearchArea.java]: Do we really want tabs for General and Hubs? We have previously decided with Owain to only have the tabs for All, Data and Information, Services and News and not to add any new ones.

Comment (giacomolm) [reply site/src/main/java/uk/nhs/digital/common/enums/SearchArea.java]: good point, this is part of the next US https://jira.digital.nhs.uk/browse/DW-223. Does it make sense to push that fix on a different branch/PR ?

Comment (jacsamell) [reply site/src/main/java/uk/nhs/digital/common/enums/SearchArea.java]: Would probably be easier to do it now as you'll end up changing exactly the same code here but it's up to you, I don't really mind 🙂

File: site/src/main/java/uk/nhs/digital/common/components/SearchComponent.java [hunk 3]
```diff
@@ -271,6 +282,35 @@ private HstQuery constructQuery(HstQueryBuilder queryBuilder, Constraint searchS
     /**
      * Publication System content types that should be included in search
      */
+    private void deriveTypes(final HstQueryBuilder query, final HstRequest request) {
```
Comment (jacsamell) [inline site/src/main/java/uk/nhs/digital/common/components/SearchComponent.java]: Just FYI I think I'm going to remove this and replace it with one single list of all the doctypes (RPS + NIL + DW) and then let the folder(/s) be the single source of filtering for the tabs. Happy to merge like this and I'll fix it later. Have discussed with @wojtek-oledzki as I know he told you the opposite originally 😉

Comment (giacomolm) [reply site/src/main/java/uk/nhs/digital/common/components/SearchComponent.java]: I am pushing the changes for DW-223 directly in this pull request, I think it's easier to sort it out here. I will update this PR shortly

Comment (jacsamell) [reply site/src/main/java/uk/nhs/digital/common/components/SearchComponent.java]: 👍

**Comment by giacomolm** [comment]
This is the list of files changed according to DW-223
	deleted:    repository-data/application/src/main/resources/hcm-content/content/documents/administration/website/search-tab-labels.yaml
	modified:   repository-data/webfiles/src/main/resources/site/freemarker/common/searchtabs.ftl
	modified:   site/src/main/java/uk/nhs/digital/common/components/SearchComponent.java
	modified:   site/src/main/java/uk/nhs/digital/common/enums/SearchArea.java

**Pull request merged**